src/mpid/ch4/subconfigure.m4

```diff
@@ -380,10 +380,16 @@ if test "$enable_ch4_shm_inline" = "yes" || test "$enable_ch4_shm_direct" = "yes
    PAC_APPEND_FLAG([-DSHM_INLINE=__shm_inline_${ch4_shm}__], [CPPFLAGS])
 fi
 
-# setup shared memory defaults
-# TODO: shm submodules should be chosen with similar configure option as that used for netmod.
-# We can add it when a shm submodule is added. Now we just simply set POSIX.
-ch4_shm=posix
+# setup shared memory submodules
+AC_ARG_WITH(ch4-shmmods, 
+    [  --with-ch4-shmmods@<:@=ARG@:>@ Specify the shared memory submodules for MPICH/CH4.
+                          Valid options are:
+                          posix        - Only enable POSIX SHM (default)
+                          xpmem:posix  - Enable both XPMEM and POSIX SHM
```

Comment:
I see this as being a problem in the future too. We probably want this to be a comma separated list, right? So you can say:

    --with-ch4-shmmods=posix,xpmem

Maybe even be able to leave out posix since it's the "fallback" for everything else.

Reply:
Yeah, comma seems more suitable here. Agree to leave `posix` out as we never allow a case where posix is not the fallback.

When future shmmods are in, we might want to use the order of arguments as the priority of each shmmod when multiple of them can optimize a communication. We do not need to consider this feature in this PR.